lib/chef/cookbook/cookbook_version_loader.rb

```diff
@@ -212,6 +224,24 @@ def chefignore
         @chefignore ||= Chefignore.new(File.basename(cookbook_path))
       end
 
+      def load_all_unignored_files
+        # List all directories that are not dotdirs, then list everything in
+        # those, *including* dotfiles and dotdirs.
+        # Finally, list all the _files_ at the root, including dotfiles, and
+        # merge those in.
+        Dir.glob(File.join(Chef::Util::PathHelper.escape_glob(cookbook_path), "*")).each do |dir|
+          dir = Chef::Util::PathHelper.cleanpath(dir)
```

Comment:
not necessary to cleanpath just to pass to File.directory? and its what you don't want with Dir.glob later